gnd/src/main/java/com/google/android/gnd/ui/basemapselector/BasemapSelectorFragment.java

```diff
@@ -50,6 +66,63 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
 
   private void onMapReady(ExtentSelector map) {
     map.renderExtentSelectionLayer();
+
+    viewModel
+        .getTiles()
+        .observe(
+            this,
+            tiles -> {
+              map.updateExtentSelections(
+                  stream(tiles).map(Extent::fromTile).collect(toImmutableSet()));
+            });
+
+    viewModel
+        .getDownloadedExtents()
+        .observe(
+            this,
+            extentId -> {
+              map.updateExtentSelections(
+                  ImmutableSet.<Extent>builder()
+                      .add(
+                          Extent.newBuilder()
+                              .setId(extentId)
+                              .setState(Extent.State.DOWNLOADED)
+                              .build())
+                      .build());
+            });
+
+    viewModel
+        .getRemovedExtents()
+        .observe(
+            this,
+            extentId -> {
+              map.updateExtentSelections(
+                  ImmutableSet.<Extent>builder()
+                      .add(Extent.newBuilder().setId(extentId).setState(Extent.State.NONE).build())
+                      .build());
+            });
+
+
+    viewModel.getExtentsPendingDownload().observe(this, extents -> {
+      if (extents.isEmpty()) {
+        downloadButton.setBackgroundTintList(ColorStateList.valueOf(getResources().getColor(R.color.colorGrey600)));
+      } else {
+        downloadButton.setBackgroundTintList(ColorStateList.valueOf(getResources().getColor(R.color.colorAccent)));
+      }
+    });
+
+    viewModel.getExtentsPendingRemoval().observe(this, extents -> {
+      if (extents.isEmpty()) {
+        removeButton.setBackgroundTintList(ColorStateList.valueOf(getResources().getColor(R.color.colorGrey600)));
+      } else {
+        removeButton.setBackgroundTintList(ColorStateList.valueOf(getResources().getColor(R.color.colorAlert)));
+      }
+    });
+
+    downloadButton.setOnClickListener(__ -> viewModel.downloadExtents());
+    removeButton.setOnClickListener(__ -> viewModel.removeExtents());
```

Comment:
I understood we'd have one button "Apply" that would apply any changes, including tiles that were toggled off (remove) and tiles that were toggled on (download). Is that not how it works here?

Reply:
Done!

Reply:
And yes, originally I had two buttons, one for each op.